compiler/rustc_data_structures/src/packed.rs

```diff
@@ -0,0 +1,71 @@
+use crate::stable_hasher::{HashStable, StableHasher};
+use rustc_serialize::{Decodable, Decoder, Encodable, Encoder};
+use std::cmp::Ordering;
+use std::fmt;
+
+#[repr(packed(8))]
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Pu128(pub u128);
```

Comment:
Is there any reason this needs to be 8-aligned? Is there anything to be gained from reducing alignment further?

Reply:
I just chose that to match its former alignment, which should still be a nice "native" alignment (on 64-bit hosts). We could try smaller, but in a quick local check, even a full `#[repr(packed)]` didn't change these `static_assert_size`s any further. Generally, I suspect there's almost always going to be a `usize` or pointer nearby keeping overall struct alignment up.

Maybe we should make it match `target_pointer_width`? But we don't test performance on other targets...